### PR TITLE
unix,win: map ESOCKTNOSUPPORT errno

### PR DESCRIPTION
**Pull request by squeek502**
This error can occur when calling `uv_socketpair` with an invalid `type`, e.g.

```c
uv_socketpair(0, 0, fds, 0, 0);
```

(the above code currently results in `Unknown system error -94` on Linux)

### CODE REVIEW

**Review by cjihrig — APPROVED** [review]

**Review by vtjnash — APPROVED** [review]
CI: https://ci.nodejs.org/job/libuv-test-commit/2148/

File: src/win/error.c [hunk 1]
```diff
@@ -168,6 +168,7 @@ int uv_translate_sys_error(int sys_errno) {
     case ERROR_NOT_SAME_DEVICE:             return UV_EXDEV;
     case ERROR_INVALID_FUNCTION:            return UV_EISDIR;
     case ERROR_META_EXPANSION_TOO_LONG:     return UV_E2BIG;
+    case WSAESOCKTNOSUPPORT:                return UV_SOCKTNOSUPPORT;
```
Comment (vtjnash) [inline src/win/error.c:171]:
```suggestion
    case WSAESOCKTNOSUPPORT:                return UV_ESOCKTNOSUPPORT;
```

**Comment by vtjnash** [comment]
https://ci.nodejs.org/job/libuv-test-commit-windows-cmake/331/

**Comment by vtjnash** [comment]
The mapping to EINVAL also needs to be deleted

**Comment by vtjnash** [comment]
https://ci.nodejs.org/job/libuv-test-commit-windows-cmake/335/

**Pull request merged**